src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -366,7 +396,7 @@ def run_safety_checker(self, image, device, dtype):
 
     def decode_latents(self, latents):
         latents = 1 / self.vae.config.scaling_factor * latents
-        image = self.vae.decode(latents).sample
+        image = self.decoder(latents).sample
```

Comment:
As explained above, so `forward` is invoked and the hook triggers.

Reply:
Instead of doing this change could we maybe just retrieve the hook from `self.vae` and call it manually. E.g. 

```py
if hasattr(self.vae, "_hf_hook") and hasattr("self.vae._hf_hook", "prev_model_hook"):
    # offload unet
```

Reply:
That would offload the unet, but will not move the vae to cuda, because we are not using the vae's `forward`.

Reply:
Ah yeah good point. Let's maybe also move it manually then there?

Reply:
I hate it too, but doing it manually is going to be ugly, I think. I'll try it.

Reply:
Another approach, which would come with its own set of difficulties, would be to teach accelerate.add_hook_to_module to hook arbitrary methods instead of hardcoding `forward`. Then ask it to put the same hook around `forward` and `decode`.

But the fact that AutoencoderKL is _already_ two separate Encoder and Decoder modules under the hood makes me wonder if there's a better way.

Reply:
I like this idea! Would be cool if we could put up a feature request for accelerate :-). This would indeed help a lot. Wdyt @pcuenca ?